Bump Spring Boot baseline from 3.2 to 3.3

ThomasVitale opened the pull request:
## Change

- Update Spring Boot baseline to 3.3.8 because 3.2.x reached EOL (2024-11-23). See: https://spring.io/projects/spring-boot#support
- Use Spring Boot Dependencies BOM in parent pom.xml and simplified dependency management and versioning.
- Configured main pipeline to run tests on Java 23 as well.
- Fix Voyage autotests when API key is not configured.

## General checklist
- [X] There are no breaking changes
- [ ] I have added unit and/or integration tests for my change
- [ ] The tests cover both positive and negative cases
- [X] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)

ThomasVitale commented:
@dliubarskyi looks like I missed a piece in the restclient module, sorry about that! It's fixed now.

I got all tests green locally. Do you have the error logs from ollama, elastic and milvus? They're all succeeding when I run the tests.

dliubarskyi commented:
@ThomasVitale thank you for fixing Ollama!

It seems that Elasticsearch and Milvus tests failed for me because I had some troubles with Podman. Now they are all green.

The only problem left seems to be failing `SpringRestClientTimeoutIT`. Does it work for you? Both tests are failing for me:
```
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0

	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:100)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:106)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:302)
	at java.base/java.util.Objects.checkIndex(Objects.java:385)
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at com.github.tomakehurst.wiremock.http.HttpServerFactoryLoader.pickMostAppropriateFrom(HttpServerFactoryLoader.java:77)
	at com.github.tomakehurst.wiremock.http.HttpServerFactoryLoader.load(HttpServerFactoryLoader.java:57)
	at com.github.tomakehurst.wiremock.WireMockServer.getHttpServerFactory(WireMockServer.java:89)
	at com.github.tomakehurst.wiremock.WireMockServer.<init>(WireMockServer.java:72)
	at dev.langchain4j.http.client.HttpClientTimeoutIT.beforeEach(HttpClientTimeoutIT.java:46)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

ThomasVitale commented:
I think I got some caching issue with Maven and IntelliJ. I could reproduce the error and spotted some other issues to run tests when not all the API keys are configured. I fixed them both.

About the HTTP timeout tests, it was a problem with WireMock not supporting Jetty 12. I fixed that here: https://github.com/langchain4j/langchain4j-spring/pull/110/files#diff-5783a7f70d37bef269301c65ee14688a069d5b7cc472cefd2ee733c7cf5ec2f6R57

dliubarskyi reviewed (approved):
@ThomasVitale thanks a lot, great job!
